### DISCUSSION TITLE
Update branch button - choose default option or choose visible options

### DISCUSSION THREAD

**Discussion by swierzbicki** [community/community#13749] [Pull Requests]
![github-update-branch](https://user-images.githubusercontent.com/7196324/160565847-510220fd-37d9-4b34-8609-730fd83398da.png)

It would be nice to have an option to choose what kind of updating branch strategy would be available inside this `update branch` button. 

In my specific case. I would like to disable `update with merging commit` because it is totally useless for `feature-branches` and it creates too much noise in feature branch git history.

**Comment by rsl** [comment]
it would also be nice to have the ability to completely remove this

**Comment by jonkoops** [comment]
+1 this is something that first time contributors see, and they think they need to keep pressing this button, causing extra notifications and CI runs. This is a bit of an anti-feature for open-source projects managing inexperienced contributors.

**Comment by nbolton** [comment]
The trick is to have a ruleset for master (include default) and a separate rulset for everything else (all branches excluding default) which forces linear history. That way, the merge button gets disabled and it defaults to rebase.

![image](https://github.com/user-attachments/assets/fe5a2450-f741-4d37-9998-e1f64d69a1a4)

![image](https://github.com/user-attachments/assets/bcae4af7-90cb-4d9f-9a86-f470460274a9)

**Reply by swierzbicki** [reply]
@nbolton i will check this. Sounds ok as a temporary solution. Because it would be easier to have simpler way to set this without doing tricks with branch protection rules 🤔

**Reply by npearson72** [reply]
In my case the opposite happens with the settings you've described. The rebase option goes away and I'm left only with merge -- which makes sense when enforcing a linear history.
